Assets/Scripts/UIManager.cs

```diff
@@ -92,12 +94,58 @@ public bool HandleLogin()
 
     public void HandleRegistration()
     {
-        // TODO: Save Character selection and name, transition to map view
+        // Preparing UserInfo object to send to server registration route
+        UserInfo userInfo = new UserInfo
+        {
+            gid = loader.GetComponent<GameManager>().gid,
+            displayName = characterName,
+            avatar = characterType,
+            addSuccessful = false
+        };
+
+        // Storing name and char type into game manager
         Debug.Log(string.Format("REGISTRATION: Character Name: {0}, Character Type: {1}", characterName, characterType));
+        loader.GetComponent<GameManager>().displayName = characterName;
         loader.GetComponent<GameManager>().charType = characterType;
-        SceneManager.LoadScene(1);
+        // Starting Coroutine to make network call for registering user
+        StartCoroutine(AddUser(userInfo));
     }
 
+    IEnumerator AddUser(UserInfo userInfo)
+    {
+        string jsonToSend = JsonUtility.ToJson(userInfo);
+        UnityWebRequest request = UnityWebRequest.Post("http://mathgo-46d6d.wl.r.appspot.com/user/registration", jsonToSend);
+
+
+        request.SetRequestHeader("Content-Type", "application/json");
+        request.SetRequestHeader("Accept", "application/json");
+        request.uploadHandler = new UploadHandlerRaw(Encoding.UTF8.GetBytes(jsonToSend));
+        yield return request.SendWebRequest();
+        var response = request.downloadHandler.text;
+
+        if (request.isNetworkError || request.isHttpError)
```

Comment:
I'm curious, does isHttpError capture the 4xx/5xx scenarios?

If so, we probably dont need to attach an addSuccessful down below and handle all non-2xxs in this block.